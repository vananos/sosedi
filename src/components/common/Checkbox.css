.checkbox-wrapper {
    display: inline-block;
    cursor: pointer;
}

.checkbox {
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100px;
    height: 100px;
    font-size: 1em;
    background-color: rgba(234, 234, 234, 1);
    color: rgba(175, 175, 175, 0.5);
    border: 10px solid rgba(234, 234, 234, 1);    
}

.checkbox input {
    display: none;
}

.checkbox span {
    text-transform: uppercase;
    font-size: 1em;
    font-weight: bold;
}

.checkbox-selected {
    background: url("../../assets/profile/check-solid.svg") rgba(222, 170, 93, 1);
    color: rgb(255, 255, 255);
    border: 10px solid rgba(244, 197, 127, 1);    
}
