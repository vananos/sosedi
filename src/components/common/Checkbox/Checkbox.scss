@import "../../../base.scss";

.checkbox {
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100px;
    height: 100px;
    background-color: $light-grey;
    color: $grey;
    border: 10px solid $light-grey;
    cursor: pointer;
    text-transform: uppercase;
    font-size: 1rem;
    font-weight: bold;

    input {
        display: none;
    }

    &-checked {
        background: url("../../../assets/profile/check-solid.svg") $brown;
        color: rgb(255, 255, 255);
        border: 10px solid $light-brown;
    }
}