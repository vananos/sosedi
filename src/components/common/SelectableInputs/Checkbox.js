import React from "react";
import SelectableInput from "./SelectableInput";

export default function Checkbox(props) {
  return <SelectableInput type="checkbox" {...props} />;
}
