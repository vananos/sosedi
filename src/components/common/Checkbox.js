import React, { Component } from "react";
import "./Checkbox.css";

export default class Checkbox extends Component {
  constructor(props) {
    super(props);
    this.state = {
      selected: props.checked
    };
  }

  changeState = () => {
    this.setState({
      selected: !this.state.selected
    });
  };

  render() {
    const { value, id, name, className, children } = this.props;
    return (
      <div className="checkbox-wrapper">
        <div
          className={`checkbox ${className ? className : ""} ${
            this.state.selected ? "checkbox-selected" : ""
          }`}
          onClick={this.changeState}
        >
          <input
            type="checkbox"
            id={id}
            name={name}
            value={value}
            checked={this.state.checked}
          />
          <span>{children}</span>
        </div>
      </div>
    );
  }
}
