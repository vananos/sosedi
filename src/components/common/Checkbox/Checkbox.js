import React, { Component } from "react";
import PropTypes from "prop-types";
import "./Checkbox.scss";

export default class Checkbox extends Component {
  constructor(props) {
    super(props);
    this.state = {
      checked: props.checked
    };
  }

  changeState = () => {
    this.setState({
      checked: !this.state.checked
    });
  };

  render() {
    const { value, id, name, children } = this.props;
    const { checked } = this.state;
    return (
      <div
        className={`checkbox ${checked ? "checkbox-checked" : ""}`}
        onClick={this.changeState}
      >
        <input
          type="checkbox"
          id={id}
          name={name}
          value={value}
          checked={checked}
          onChange={() => {}}
        />
        {children}
      </div>
    );
  }
}

Checkbox.defaultProps = {
  checked : false
};

Checkbox.propTypes = {
  value: PropTypes.string.isRequired,
  id: PropTypes.string,
  name: PropTypes.string.isRequired,
  value: PropTypes.string,
  checked: PropTypes.bool,
  children: PropTypes.node
};
