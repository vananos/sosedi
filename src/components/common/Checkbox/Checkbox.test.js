import { shallow } from "enzyme";
import React from "react";
import Checkbox from "./Checkbox";

describe("Checkbox", () => {
  const checkboxProps = () => ({
    name: "test-checkbox",
    value: "test?"
  });

  it("checkbox is not checked, click -> checked", () => {
    const checkboxWrapper = shallow(<Checkbox {...checkboxProps()} />);
    checkboxWrapper
      .find("div")
      .first()
      .simulate("click");

    expect(checkboxWrapper).toMatchSnapshot();
  });

  it("checkbox is checked, click -> not checked", () => {
    const checkboxWrapper = shallow(
      <Checkbox {...checkboxProps()} checked={true} />
    );
    
    checkboxWrapper
      .find("div")
      .first()
      .simulate("click");

    expect(checkboxWrapper).toMatchSnapshot();
  });
});
