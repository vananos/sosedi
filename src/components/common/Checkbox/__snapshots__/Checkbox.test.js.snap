// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox checkbox is checked, click -> not checked 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Checkbox
    checked={true}
    name="test-checkbox"
    value="test?"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          checked={false}
          name="test-checkbox"
          onChange={[Function]}
          type="checkbox"
          value="test?"
        />,
        undefined,
      ],
      "className": "checkbox ",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "checked": false,
          "id": undefined,
          "name": "test-checkbox",
          "onChange": [Function],
          "type": "checkbox",
          "value": "test?",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            checked={false}
            name="test-checkbox"
            onChange={[Function]}
            type="checkbox"
            value="test?"
          />,
          undefined,
        ],
        "className": "checkbox ",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "checked": false,
            "id": undefined,
            "name": "test-checkbox",
            "onChange": [Function],
            "type": "checkbox",
            "value": "test?",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Checkbox checkbox is not checked, click -> checked 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Checkbox
    checked={false}
    name="test-checkbox"
    value="test?"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          checked={true}
          name="test-checkbox"
          onChange={[Function]}
          type="checkbox"
          value="test?"
        />,
        undefined,
      ],
      "className": "checkbox checkbox-checked",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "checked": true,
          "id": undefined,
          "name": "test-checkbox",
          "onChange": [Function],
          "type": "checkbox",
          "value": "test?",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            checked={true}
            name="test-checkbox"
            onChange={[Function]}
            type="checkbox"
            value="test?"
          />,
          undefined,
        ],
        "className": "checkbox checkbox-checked",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "checked": true,
            "id": undefined,
            "name": "test-checkbox",
            "onChange": [Function],
            "type": "checkbox",
            "value": "test?",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
